python/ray/tune/integration/wandb.py

```diff
@@ -95,18 +95,26 @@ def _set_api_key(wandb_config):
     api_key_file = os.path.expanduser(wandb_config.pop("api_key_file", ""))
     api_key = wandb_config.pop("api_key", None)
 
-    if api_key_file:
+    try:
+        # Check if user is already logged into wandb.
+        wandb.ensure_configured()
+        if wandb.api.api_key:
+            logger.info("Already logged into W&B.")
+            return
+    except AttributeError:
+        if api_key_file:
+            if api_key:
+                raise ValueError("Both WandB `api_key_file` and `api_key` set.")
```

Comment:
shouldn't this happen before the try-catch?

Reply:
I think we will have to discuss the login priorities. Right now, wandb login takes priority over the other methods, however, I think if someone explicitly passes an `api_key_file` or even `api_key` in the config, he would like to log to that account.

IMO, the priority should be like this:

1) Config `api_key`/`api_key_file` (only one allowed)
2) Environment variable
3) wandb login

Reply:
Makes sense. I updated it to reflect that logic.